grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1089,6 +1094,106 @@ class StockMovementService {
+                    'code',
+                    'name',
+                    'lot',
+                    'expiration',
+                    'binLocation',
+                    'quantity',
+            ]
+            return csvMapReader.toList()
+
+        } catch (Exception e) {
+            throw new RuntimeException("Error parsing order item CSV: " + e.message, e)
+        }
+    }
+
+    void validatePicklistListImport(List<Map> data, List<PickPageItem> pickPageItems) {
+        data?.eachWithIndex { Map params, index ->
+            params.errors = new ArrayList()
+            if (!params.id) {
+                params.errors.push("Requisition item id is required")
+            }
+            if (params.quantity == null) {
+                params.errors.push("Quantity picked are required")
+            }
+            if (params.lot?.contains("E") && NumberUtils.isNumber(params.lot)) {
+                params.errors.push("Lot numbers must not be specified in scientific notation. " +
+                        "Please reformat field with Lot Number: \"${params.lot}\" to a number format")
+            }
+
+            PickPageItem pickPageItem = pickPageItems.find { it.requisitionItem?.id == params.id }
+            if (!pickPageItem) {
+                params.errors.push("Requisition item id: ${params.id} not found")
+            }
+            if (pickPageItem) {
+                AvailableItem availableItem = pickPageItem.availableItems?.find { item ->
+                    Boolean binLocationMatches = params.binLocation ? item.binLocation?.name == params.binLocation : !item.binLocation
+                    Boolean lotMatches = params.lot == item.inventoryItem?.lotNumber
+                    binLocationMatches && lotMatches
+                }
+
+                if (!availableItem) {
+                    params.errors.push("There is no item available with: ${params.lot ? "lot ${params.lot}" : ""} ${params.binLocation ? "bin ${params.binLocation}" : ""}")
+                }
+            }
+        }
+
+        Boolean isUnderPickAllowed = Holders.config.openboxes.picklist.underpick.enabled
+        Boolean isOverPickAllowed = Holders.config.openboxes.picklist.overpick.enabled
+
+        data.groupBy { it.id }.each { itemId, params ->
+            PickPageItem pickPageItem = pickPageItems.find { it.requisitionItem?.id == itemId }
+            Integer itemQuantitySum = params.sum { Integer.parseInt(it.quantity) }
+            if (!isOverPickAllowed && pickPageItem && itemQuantitySum > pickPageItem.requisitionItem.quantity) {
+                errors.push("Item ${itemId} is overpicked, expedted quantity ${pickPageItem.requisitionItem.quantity}, received ${itemsSum}")
+            }
+            if (!isUnderPickAllowed && pickPageItem && itemQuantitySum < pickPageItem.requisitionItem.quantity) {
+                errors.push("Item ${itemId} is underpicked, expedted quantity ${pickPageItem.requisitionItem.quantity}, received ${itemsSum}")
```

Comment:
I've just noticed a typo here and in the message above -> `expedted`

Reply:
I noticed that as well, but forgot to comment.